lib/src/package_config_json.dart

```diff
@@ -196,37 +244,55 @@ PackageConfig parsePackageConfigJson(dynamic json, Uri baseLocation) {
           break;
       }
     });
-    if (name == null) {
-      throw PackageConfigFormatException("Missing name entry", entry);
+    if (!hasName) {
+      onError(PackageConfigFormatException("Missing name entry", entry));
     }
-    if (rootUri == null) {
-      throw PackageConfigFormatException("Missing rootUri entry", entry);
+    if (!hasRoot) {
+      onError(PackageConfigFormatException("Missing rootUri entry", entry));
     }
+    if (name == null || rootUri == null) return null;
     Uri root = baseLocation.resolve(rootUri);
-    Uri /*?*/ packageRoot = root;
+    Uri packageRoot = root;
     if (packageUri != null) packageRoot = root.resolve(packageUri);
-    try {
-      return SimplePackage(name, root, packageRoot, languageVersion, extraData);
-    } on ArgumentError catch (e) {
-      throw PackageConfigFormatException(e.message, e.invalidValue);
+
+    LanguageVersion /*?*/ version;
+    if (languageVersion != null) {
```

Comment:
languageVersion is null if it's not a string in the json format (line 240 --- for some reason I cannot mark that line).